sdcm/nemesis.py

```diff
@@ -1725,13 +1727,44 @@ def disrupt_nodetool_cleanup(self):
             for keyspace in test_keyspaces:
                 node.run_nodetool(sub_cmd="cleanup", args=keyspace)
 
-    def _prepare_test_table(self, ks='keyspace1', table=None):
+    def _prepare_test_table(self,
+                            ks: str = 'keyspace1',
+                            table: str | None = None,
+                            pre_create_objects: bool = False) -> None:
         ks_cfs = self.cluster.get_non_system_ks_cf_list(db_node=self.target_node)
         table_exist = f'{ks}.{table}' in ks_cfs if table else True
 
         test_keyspaces = self.cluster.get_test_keyspaces()
-        # if keyspace or table doesn't exist, create it by cassandra-stress
+        # If keyspace or table doesn't exist, create it
         if ks not in test_keyspaces or not table_exist:
+            # If this option is True we create ks and table in advance and let the C-S thread only fill them with data,
+            # otherwise ks and table are also created by C-S thread.
+            if pre_create_objects:
+                self.log.debug("Going to create a new keyspace `%s'...", ks)
+                self.tester.create_keyspace(
+                    keyspace_name=ks,
+                    replication_factor=self.tester.reliable_replication_factor,
+                )
+
+                self.cluster.wait_for_schema_agreement()
+
+                table_name = table or "standard1"
```

Comment:
instead of having it here, we could have this default set in the function signature

Reply:
Yes, I could, but in this case, I would need to change how this method is called in `disrupt_truncate`.
We have there: `self._prepare_test_table(ks=keyspace_truncate)`, I would need to change it to `self._prepare_test_table(ks=keyspace_truncate, table=None)` in order to not affect anything.
Should I do this?

Reply:
but  this is exactly as having it in the method signature... if `table` will be `None`, then it will be assigned `standard1`, but if you pass anything else, then it will be `table`...
if the function call will be `_prepare_test_table(ks: str = 'keyspace1', table: str = 'standard1' ....)`

Reply:
As agreed, the only thing we need in this workaround is to wait for the schema agreement in `disrupt_no_corrupt_repair`.
The proposed change is not needed.